src/core.js

```diff
@@ -103,10 +103,21 @@ export default class Core {
   }
 
   on(query, events, callback) {
-    const el = _element(query)
-    if (!el) return
+    let elements = _elements(query)
+    if (!elements) return
+
+    if (elements.length == undefined) elements = [elements]
 
-    events.split(' ').forEach(event => el.addEventListener(event, callback))
+    elements.forEach(component => {
+      events.split(' ').forEach(event => component.addEventListener(event, callback))
+    })
+  }
+
+  _elements(query) {
```

Comment:
all current "private" methods are now defined in the bottom, I think this one should live near `_element`